### PR TITLE
Origin/develop

### PR DESCRIPTION
**Pull request by ksatya77**

### CODE REVIEW

File: interop-switch/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "interop-switch.fullname" . }}
+  labels:
+    app: {{ template "interop-switch.name" . }}
```
Comment (mdebarros) [inline interop-switch/templates/deployment.yaml]: We are using fullname for the app label.
Please change too: app: {{ template "interop-switch.fullname" . }}

File: interop-switch/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: {{ template "interop-switch.name" . }}
```
Comment (mdebarros) [inline interop-switch/templates/deployment.yaml]: We are using fullname for the app label.
Please change too: app: {{ template "interop-switch.fullname" . }}

File: interop-switch/templates/ingress.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{{- if .Values.ingress.enabled -}}
+{{- $serviceName := include "interop-switch.fullname" . -}}
+{{- $servicePort := .Values.service.externalPort -}}
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: {{ template "interop-switch.fullname" . }}
+  labels:
+    app: {{ template "interop-switch.name" . }}
```
Comment (mdebarros) [inline interop-switch/templates/ingress.yaml]: We are using fullname for the app label.
Please change too: app: {{ template "interop-switch.fullname" . }}

File: interop-switch/templates/service.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "interop-switch.fullname" . }}
+  labels:
+    app: {{ template "interop-switch.name" . }}
```
Comment (mdebarros) [inline interop-switch/templates/service.yaml]: We are using fullname for the app label.
Please change too: app: {{ template "interop-switch.fullname" . }}

File: interop-switch/templates/service.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    heritage: {{ .Release.Service }}
+spec:
+  type: {{ .Values.service.type }}
+  ports:
+    - port: {{ .Values.service.externalPort }}
+      targetPort: {{ .Values.service.internalPort }}
+      protocol: TCP
+      name: {{ .Values.service.name }}
+  selector:
+    app: {{ template "interop-switch.name" . }}
```
Comment (mdebarros) [inline interop-switch/templates/service.yaml]: We are using fullname for the app label.
Please change too: app: {{ template "interop-switch.fullname" . }}

File: interop-switch/values.yaml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+## Optionally specify an array of imagePullSecrets.
+## Secrets must be manually created in the namespace.
+## ref: https://kubernetes.io/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod
+##
+#  imagePullSecrets:
+#    - name: myregistrykey
+  pullPolicy: IfNotPresent
+
+config:
+  SERVICE_8088_NAME: 'interop-switch'
```
Comment (mdebarros) [inline interop-switch/values.yaml]: I do not see this being mapped in the config. Is it being used?

File: interop-switch/values.yaml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  MAX_JVM_MEMORY: '256'
```
Comment (mdebarros) [inline interop-switch/values.yaml]: I do not see this being mapped in the config. Is it being used?

File: README.md [hunk 8]
```diff
@@ -162,6 +162,75 @@ This will deploy a new Ingress Nginx Controller with TCP and HTTP Ingress
 ### Package
 1.) Package charts
 
-Package all charts, and created an index.yaml in ./repo directory
```
Comment (mdebarros) [inline README.md]: Why was this removed?

File: README.md [hunk 9]
```diff
@@ -162,6 +162,75 @@ This will deploy a new Ingress Nginx Controller with TCP and HTTP Ingress
...
+<<<<<<< HEAD
```
Comment (mdebarros) [inline README.md]: Looks like you had a bad merge. Please clean this up. I suspect this removed line 165.

**Comment by mdebarros** [comment]
Please @ksatya77 re-base & squash-merge your commits.

File: interop-switch/templates/deployment.yaml (hunk 2 above)
Comment (ksatya77) [reply interop-switch/templates/deployment.yaml]: @mdebarros I believe I made all the changes you suggested. Added some additional functionality. Can you please review.
